tests/test_readme.py

```
@@ -0,0 +1,354 @@
+        has_markdown_link = bool(re.search(r'\[.*\]\(.*charlielabs\.ai.*\)', content, re.IGNORECASE))
+        has_direct_link = bool(re.search(r'https?://.*charlielabs\.ai', content, re.IGNORECASE))
+        
+        assert has_markdown_link or has_direct_link, \
+            "Charlie Labs URL must be either a markdown link or direct URL"
+    
+    def test_no_broken_markdown_links(self):
+        """Test that markdown link syntax is not broken"""
+        content = self.get_readme_content()
+        
+        # Check for broken link patterns
+        broken_patterns = [
```

Ruff (0.15.2)

[error] Local variable `broken_patterns` is assigned to but never used

Remove assignment to unused variable `broken_patterns`

(F841)